1.process_data/nbconverted/single_cell_qc.py

```diff
@@ -0,0 +1,277 @@
+        "Image_FileName_OrigDNA",
+    ]
+]
+
+
+print(irregular_nuclei_outliers_cdf.shape)
+irregular_nuclei_outliers_cdf.sort_values(by="Nuclei_AreaShape_FormFactor", ascending=False).head(2)
+
+
+# ### Randomly sample outlier rows to visually inspect if the threshold looks to be optimized
+
+# In[7]:
+
+
+irregular_nuclei_outliers_cdf.sample(n=5, random_state=0)
+
+
+# ### Generate a new dataframe to save the original indices for filtering prior to further preprocessing
+
+# In[8]:
+
+
+# Create a new dataframe for failed QC indices
+failed_qc_indices = irregular_nuclei_outliers[metadata_cols].copy()
+failed_qc_indices["original_index"] = failed_qc_indices.index  # Store original index
+failed_qc_indices = failed_qc_indices.reset_index(drop=True)  # Reset index
+failed_qc_indices["cqc.formfactor_displacement_outlier"] = True
+
+print(failed_qc_indices.shape)
+failed_qc_indices.head()
+
+
+# 2. Nuclei segmentations with holes and also irregular shaped
+# 
+# We need to include this extra condition as it was discovered that there were more poorly segmented cells not caught, especially those over-segmented nuclei that contained holes, which is not common for a nuclei segmentation.
+# 
+# - `Compactness`, which detects irregular nuclei and nuclei containing holes.
+
+# ### Detect outliers and show single-cell image crops with CytoDataFrame (cdf)
+
+# In[9]:
+
+
+# find mis-segmented nuclei due using thresholds that maximizes
+# removing most technical outliers and minimizes good cells
+feature_thresholds = {
+    "Nuclei_AreaShape_Compactness": 4.05,
+}
+
+poor_nuclei_outliers = cosmicqc.find_outliers(
+    df=df_merged_single_cells,
+    metadata_columns=metadata_cols,
+    feature_thresholds=feature_thresholds
+)
+
+poor_nuclei_outliers_cdf = CytoDataFrame(
+    data=pd.DataFrame(poor_nuclei_outliers),
+    data_context_dir=f"/media/jenna/8TB-C/work/JUMP-download-images/JUMP-single-cell/0.download_data/data/images/{plate_id}/orig",
+    data_outline_context_dir=f"/media/jenna/8TB-C/work/JUMP-download-images/JUMP-single-cell/0.download_data/data/images/{plate_id}/outlines",
+    segmentation_file_regex=outline_to_orig_mapping,
+)[
+    [
+        "Nuclei_AreaShape_Compactness",
+        "Image_FileName_OrigDNA",
+    ]
+]
+
+
+print(poor_nuclei_outliers_cdf.shape)
+poor_nuclei_outliers_cdf.sort_values(by="Nuclei_AreaShape_Compactness", ascending=True).head(2)
+
+
+# ### Randomly sample outlier rows to visually inspect if the threshold looks to be optimized
+
+# In[10]:
+
+
+poor_nuclei_outliers_cdf.sample(n=5, random_state=0)
+
+
+# ### Save the original indices for failed single cells to compressed CSV files
+
+# In[11]:
+
+
+# Create a new dataframe for poor nuclei outliers
+poor_qc_indices = poor_nuclei_outliers[metadata_cols].copy()
+poor_qc_indices["original_index"] = poor_qc_indices.index  # Store original index
+poor_qc_indices = poor_qc_indices.reset_index(drop=True)  # Reset index
+poor_qc_indices["cqc.compactness_outlier"] = True
```

Comment:
Minor suggestion for greater clarity on what the boolean stipulates.
```suggestion
poor_qc_indices["cqc.compactness_outlier.is_outlier"] = True
```

Reply:
Great point! I will do that now!